File: Polymorphism (METHOD OVERLOADING AND OVERRIDDING).cpp
#include <iostream>
#include <string>
using namespace std;

class Country {
public:
    string name;

    Country(string n) {
        this->name = n;
    }

    virtual void language() {
        cout << "Language is English" << endl;
    }
};

class Bangladesh : public Country {
public:
    Bangladesh(string n) : Country(n) {
     
    }

    // method overriding
    void language() override {
        cout << "Language is Bangla" << endl;
    }
};

class India : public Country {
public:
    India(string n) : Country(n) {}

    // method overloading
    void language(string a, string b) {
        cout << "Language " << a << " and " << b << endl;
    }
};

int main() {
    India a("India");

    a.language("Hindi", "Bangla");

    return 0;
}

File: Polymorphism (METHOD OVERLOADING AND OVERRIDDING).py
class Country:
    
    name = ''
    def __init__(self, n) -> None:
        self.name = n

    def language(self):
        print('Language is Englishh')


class Bangladesh(Country):
    def __init__(self, n) -> None:
        super().__init__(n)
#method overriding
    def language(self):
        print('language is bangla')

class India(Country):
    def __init__(self, n) -> None:
        super().__init__(n)
#method overloading
    def language(self,a,b):
        print(f'language {a} and {b}')     


a = India('India')  

a.language('Hindi','Bangla')